### DISCUSSION TITLE
Force run and pause Ent Periodic Jobs

### DISCUSSION THREAD

**Discussion by danilo-silva-cw** [sidekiq/sidekiq#5901] [Show and tell]
Hello @mperham, 

The features are force run and pause cron of a job using the `LID` as identifier. 

![Screenshot from 2023-04-28 10-06-06](https://user-images.githubusercontent.com/97977377/235157692-c5e3a7d7-07c3-4c1f-a44e-87f8a5e12ae7.png)

I created a private gem using Ruby prepends to add the functionality of pausing or starting a cron job based on the issue https://github.com/sidekiq/sidekiq/issues/5000. The only part of your gem `sidekiq-ent` that I had to use was the `loop.erb` view file to add the buttons.

I would like to publish this gem as a opensource, and in a future if you want to incorporate to `sidekiq-ent`

Would you allow this publication? Or if you prefer, may I send you the gem through some means so that you can better review it?

**Comment by mperham** [comment]
That's neat that you were able to build this!

I've had concerns about the Pause functionality. If someone forgets they pressed it or doesn't tell another team member, it could cause a lot of confusion about why the cron job is not running.

I don't think you can publish the gem without republishing a modified `loop.erb`, which isn't allowed by the license. Do you want to send me a patch so you can add your table columns without modifying and distributing `loop.erb`? Sidekiq Pro does something similar so it can inject the filter input boxes, Sidekiq provides an [empty impl](https://github.com/sidekiq/sidekiq/blob/main/web/views/retries.erb#L6) which Pro overrides.

**Comment by stevenharman** [comment]
The pause functionality aside, I too have wanted a "run right now" button in the UI. Lacking that, I tend to just fire up a console session and enqueue one myself. Which works, but isn't as convenient as the UI. Especially for local testing/debugging. So, a nice-to-have, for me. 😄

**Reply by dannnylo** [reply]
Hello @stevenharman, 
I finally managed to launch my gem, with this functionality.
 
https://rubygems.org/gems/sidekiq-belt

**Comment by danilo-silva-cw** [comment]
Hello @mperham.

Sorry for the delay in reply.
I sent via email the gist link with the diff of the view with the subject "Discussions #5901 Force run and pause Ent Periodic Jobs".

I am waiting for your reply.

Thanks

**Reply by mperham** [reply]
I think my long-term plan here is to allow third-party code to add columns to all tables in the Web UI. I'll need to rejigger the Web UI to allow this but it's a common request from people looking to enhance the data in various tables with links or buttons. Thanks for sending me your code as one example.

I don't have a timeline for this work but I'm taking a sabbatical this summer so look for progress in the fall.

**Comment by dannnylo** [comment]
I created a simple way to inject buttons into views.
https://github.com/dannnylo/sidekiq-belt/blob/main/lib/sidekiq/web_action_helper.rb#L12
https://github.com/dannnylo/sidekiq-belt/blob/main/lib/sidekiq/web_router_helper.rb#L9

And to use is like https://github.com/dannnylo/sidekiq-belt/blob/main/lib/sidekiq/belt/ent/periodic_pause.rb#L39

```
app.replace_content("/loops") do |content|
    content.gsub!("</th>\n    </tr>", "</th><th><%= t('Pause/Unpause') %></th></th>\n    </tr>")
end
```
To make it easy and reusable at the end of Sidekiq views could have a comment as a tag to UI replacement.
```
<table>
  <thead>
    <th></th>
  </thead>
  <tbody>
    <tr>
      <td></td>
      <!-- END_COLUMN -->
    </tr>
    <!-- END_LINE -->
  </tbody>
</table>
```
